lib/iris/analysis/_area_weighted.py

```diff
@@ -97,6 +101,6 @@ def __call__(self, cube):
                 "The given cube is not defined on the same "
                 "source grid as this regridder."
             )
-        return eregrid.regrid_area_weighted_rectilinear_src_and_grid(
-            cube, self._target_grid_cube, mdtol=self._mdtol
+        return eregrid._regrid_area_weighted_rectilinear_src_and_grid__perform(
+            cube, self._regrid_info, mdtol=self._mdtol
```

Comment:
We need to have checks in here to ensure that the incoming src_cube coordinate info, matches that of the original src_cube used to generate the weights.  This latter information will now be in the _regrid_info variables.

Reply:
Done: [27a4add](https://github.com/ehogan/iris/commit/27a4addcdde174fa74285bccb957843847cb5982), with test: [ec9045f](https://github.com/SciTools/iris/pull/3606/commits/ec9045f2c131a4f2248c2c8f0dbd5610faa86976)